File: php/src/Model/SupervisorCrewMember.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * One crew member's current load.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class SupervisorCrewMember
{
    public function __construct(
        public readonly string $workerId,
        public readonly int $backlog,
        public readonly bool $available,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            backlog: (int) ($data['backlog'] ?? 0),
            available: (bool) ($data['available'] ?? false),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/SupervisorCrewMember.java
package io.fivexer.sdk.model;

/**
 * One crew member's current load.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class SupervisorCrewMember {
    private String workerId;
    private Integer backlog;
    private Boolean available;

    public String getWorkerId() { return workerId; }
    public Integer getBacklog() { return backlog; }
    public Boolean getAvailable() { return available; }
}
